lib/libspl/include/assert.h

```diff
@@ -56,10 +56,12 @@ do {									\
 	const TYPE __left = (TYPE)(LEFT);				\
 	const TYPE __right = (TYPE)(RIGHT);				\
 	if (!(__left OP __right)) {					\
-		char *__buf = alloca(256);				\
-		(void) snprintf(__buf, 256,				\
+		char *__buf = NULL;					\
+		int rc = asprintf(&__buf,				\
```

Comment:
The other four callers of `libspl_assert()` just pass it a const string, so it makes sense to have `VERIFY3_IMPL` pass in a string as well (rather than a format string + vars).  

I'll get rid of the `asprintf` and use a "snprintf to get size + alloca(size)" to allocate `__buf`.